test/test-util.js

```diff
@@ -6,203 +6,160 @@
-  assert.equal(libUtil.urlParse("a"), null);
-  assert.equal(libUtil.urlParse("a/b"), null);
-  assert.equal(libUtil.urlParse("a//b"), null);
-  assert.equal(libUtil.urlParse("/a"), null);
-  assert.equal(libUtil.urlParse("data:foo,bar"), null);
-
-  let parsed = libUtil.urlParse("http://x-y.com/bar");
-  assert.equal(parsed.scheme, "http");
-  assert.equal(parsed.host, "x-y.com");
-  assert.equal(parsed.path, "/bar");
-
-  const webpackURL = "webpack:///webpack/bootstrap 67e184f9679733298d44";
-  parsed = libUtil.urlParse(webpackURL);
-  assert.equal(parsed.scheme, "webpack");
-  assert.equal(parsed.host, "");
-  assert.equal(parsed.path, "/webpack/bootstrap 67e184f9679733298d44");
-  assert.equal(webpackURL, libUtil.urlGenerate(parsed));
-};
-
 exports["test normalize()"] = function(assert) {
   assert.equal(libUtil.normalize("/.."), "/");
   assert.equal(libUtil.normalize("/../"), "/");
   assert.equal(libUtil.normalize("/../../../.."), "/");
   assert.equal(libUtil.normalize("/../../../../a/b/c"), "/a/b/c");
   assert.equal(libUtil.normalize("/a/b/c/../../../d/../../e"), "/e");
 
-  assert.equal(libUtil.normalize(".."), "..");
+  assert.equal(libUtil.normalize(".."), "../");
   assert.equal(libUtil.normalize("../"), "../");
+
   assert.equal(libUtil.normalize("../../a/"), "../../a/");
-  assert.equal(libUtil.normalize("a/.."), ".");
-  assert.equal(libUtil.normalize("a/../../.."), "../..");
+  assert.equal(libUtil.normalize("a/.."), "");
+  assert.equal(libUtil.normalize("a/../../.."), "../../");
 
   assert.equal(libUtil.normalize("/."), "/");
   assert.equal(libUtil.normalize("/./"), "/");
   assert.equal(libUtil.normalize("/./././."), "/");
   assert.equal(libUtil.normalize("/././././a/b/c"), "/a/b/c");
   assert.equal(libUtil.normalize("/a/b/c/./././d/././e"), "/a/b/c/d/e");
 
-  assert.equal(libUtil.normalize(""), ".");
-  assert.equal(libUtil.normalize("."), ".");
-  assert.equal(libUtil.normalize("./"), ".");
+  assert.equal(libUtil.normalize(""), "");
+  assert.equal(libUtil.normalize("."), "");
+  assert.equal(libUtil.normalize("./"), "");
   assert.equal(libUtil.normalize("././a"), "a");
   assert.equal(libUtil.normalize("a/./"), "a/");
-  assert.equal(libUtil.normalize("a/././."), "a");
+  assert.equal(libUtil.normalize("a/././."), "a/");
 
-  assert.equal(libUtil.normalize("/a/b//c////d/////"), "/a/b/c/d/");
-  assert.equal(libUtil.normalize("///a/b//c////d/////"), "///a/b/c/d/");
-  assert.equal(libUtil.normalize("a/b//c////d"), "a/b/c/d");
+  assert.equal(libUtil.normalize("/a/b//c////d/////"), "/a/b//c////d/////");
 
-  assert.equal(libUtil.normalize(".///.././../a/b//./.."), "../../a");
+  assert.equal(libUtil.normalize("///a/b//c////d/////"), "//a/b//c////d/////");
+  assert.equal(libUtil.normalize("a/b//c////d"), "a/b//c////d");
 
-  assert.equal(libUtil.normalize("http://www.example.com"), "http://www.example.com");
+  assert.equal(libUtil.normalize(".///.././../a/b//./.."), "a/b/");
```

Comment:
The original result here seems more correct than this one to me.  What's the rationale for this change?

Reply:
Similar to above, the slashes have meaning.
```
new URL(".///.././../a/b//./..", "http://example.com/1/2/3/4/").href
// "http://example.com/1/2/3/4/a/b/"
```
so the path is `a/b/` relative to the base.